pkg/controller/postgresql.go

```diff
@@ -37,6 +51,11 @@ func (c *Controller) clusterListFunc(options api.ListOptions) (runtime.Object, e
 		return nil, fmt.Errorf("could not extract list of postgresql objects: %v", err)
 	}
 
+	if time.Since(c.lastClusterSyncTime) <= c.opConfig.ResyncPeriod {
+		c.logger.Debugln("skipping resync of clusters")
```

Comment:
The problem with this approach is that in an extreme case it would take close to 2 * lastClusterSyncTime before the sync runs (if the informer launched that function in minute 1, the go routine called it on minute 5, bailed out and then called it for the second time at minute 10, given that the sync interval is 5 minutes).